### PR TITLE
Adapt website to display dromaeo results.

### PR DESCRIPTION
**Pull request by JayNakrani**
This pull modifies the db tables and websites to display the results of Dromaeo test suite on website.
I have deployed an experimental version of this on Amazon EC2. You can see it in action at [ec2-52-88-122-83.us-west-2.compute.amazonaws.com](http://ec2-52-88-122-83.us-west-2.compute.amazonaws.com/)

Currently Overview and Regression sections don't work on the site. So I have removed their links from home page. Will add once they are working. I am planning to first add a few more benchmarks and then take care of regression and overview parts.

@h4writer, @jdm: Please give your thoughts on these changes.

@jdm: How are we planning to go for actual hosting of this site? Right now I am using Amazon's free tier, but that is limited. Soon I will run out of my free quota!

### CODE REVIEW

**Comment by JayNakrani** [comment]
Oh, yes. One more thing.
Currently I only ran Dromaeo's [3D tests](http://dromaeo.com/?3d) since they're the shortest on the website. And hence we only see 3 sub-tests. In actual deployment, we will be running the [recommended tests](http://dromaeo.com/?recommended).

**Comment by h4writer** [comment]
What about hosting these results on AWFY itself? I can give you the submitter url and make a custom front that shows the servo results, like you have now. Seems like this is appropriate to centralize this.

**Comment by JayNakrani** [comment]
Sounds good to me. :smiley:

**Comment by JayNakrani** [comment]
While we are at it, we can reuse the same slaves for servo's benchmarking as well.

File: database/schema.sql [hunk 1]
```diff
@@ -425,7 +426,8 @@ INSERT INTO `awfy_vendor` (`id`, `name`, `vendor`, `csetURL`, `browser`, `rangeU
 (11, 'SpiderMonkey', '1GB inbound', 'http://hg.mozilla.org/integration/mozilla-inbound/rev/', 'Firefox OS', 'http://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?fromchange={from}&tochange={to}'),
 (12, 'SpiderMonkey', '512MB inbound', 'http://hg.mozilla.org/integration/mozilla-inbound/rev/', 'Firefox OS', 'http://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?fromchange={from}&tochange={to}'),
 (13, 'SpiderMonkey', '256MB inbound', 'http://hg.mozilla.org/integration/mozilla-inbound/rev/', 'Firefox OS', 'http://hg.mozilla.org/integration/mozilla-inbound/pushloghtml?fromchange={from}&tochange={to}'),
-(14, 'V8', 'Google', 'https://chromium.googlesource.com/v8/v8/+/', 'Chrome', 'http://code.google.com/p/v8/source/list?num={num}&start={to}');
+(14, 'V8', 'Google', 'https://chromium.googlesource.com/v8/v8/+/', 'Chrome', 'http://code.google.com/p/v8/source/list?num={num}&start={to}'),
+(15, 'SpiderMonkey', 'Mozilla', 'https://github.com/servo/servo/commits', 'Servo', 'https://github.com/servo/servo/compare/{from}...{to}');
```
Comment (h4writer) [inline database/schema.sql:431]: I would prefer to do this using the migration scripts now.
-> insert vendor, retrieve ID
-> insert mode with right vendor_id

migration scripts are quite easy. Just a php file containing the function migrate(), which should migrate to the new version. And rollback() which should undo the migrate() function. Like:
https://github.com/h4writer/arewefastyet/blob/demotestamptosortorder/database/migration-1.php

**Comment by jdm** [comment]
@dhananjay92 I'll ask that question in the meeting today!

**Comment by jdm** [comment]
Actually, if we can reuse AWFY then that's even easier!

**Comment by h4writer** [comment]
@dhananjay92 I've added you to google chat to talk about how to push to awfy etc. I didn't find you in the irc channel #servo.

**Comment by JayNakrani** [comment]
@h4writer I just joined channel #servo. My nick is `Dhananjay`. We can continue our discussion about pushing to awfy there.

File: database/schema.sql (hunk 1 above)
Comment (JayNakrani) [inline database/schema.sql:431]: Yup. Changed it to migration. Now that change is in `migration-1.php`.

**Comment by JayNakrani** [comment]
@h4writer Any more comments on this? Should I merge it?

File: database/migration-1.php [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+function migrate() {
+    // INSERT an entry into awfy_vendor
+    mysql_query("INSERT INTO `awfy_vendor` (`id`, `name`, `vendor`, `csetURL`, `browser`, `rangeURL`) VALUES (15, 'SpiderMonkey', 'Mozilla', 'https://github.com/servo/servo/commits', 'Servo', 'https://github.com/servo/servo/compare/{from}...{to}');") or die(mysql_error());
+
+    // Get last inserted id
+    $vendor_id = mysql_insert_id();
+
+    // INSERT into awfy_mode with inserted vendor id
+    mysql_query("INSERT INTO `awfy_mode` (`id`, `vendor_id`, `mode`, `name`, `color`, `level`) VALUES (46, {$vendor_id}, 'servo', 'Servo', '#FF0000', 1);") or die(mysql_error());
+}
```
Comment (h4writer) [inline database/migration-1.php:17]: Please don't hardcode the id

File: database/migration-1.php [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<?php
+// Add entries for Servo in the awfy_vendor and awfy_mode.
+function migrate() {
+    // INSERT an entry into awfy_vendor
+    mysql_query("INSERT INTO `awfy_vendor` (`id`, `name`, `vendor`, `csetURL`, `browser`, `rangeURL`) VALUES (15, 'SpiderMonkey', 'Mozilla', 'https://github.com/servo/servo/commits', 'Servo', 'https://github.com/servo/servo/compare/{from}...{to}');") or die(mysql_error());
```
Comment (h4writer) [inline database/migration-1.php]: Please don't hardcode the id

File: database/migration-1.php [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    // Get last inserted id
+    $vendor_id = mysql_insert_id();
+
+    // INSERT into awfy_mode with inserted vendor id
+    mysql_query("INSERT INTO `awfy_mode` (`id`, `vendor_id`, `mode`, `name`, `color`, `level`) VALUES (46, {$vendor_id}, 'servo', 'Servo', '#FF0000', 1);") or die(mysql_error());
+}
+
+function rollback() {
+    // Delete mode first
+    mysql_query("DELETE FROM `awfy_mode` WHERE `awfy_mode`.`id` = 46") or die(mysql_error());
```
Comment (h4writer) [inline database/migration-1.php]: Don't use the id to remove it, but check for some unique data like e.g. browser to delete the right row.

File: database/migration-1.php [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+    // Delete vendor now
+    mysql_query("DELETE FROM `awfy_vendor` WHERE `awfy_vendor`.`id` = 15") or die(mysql_error());
```
Comment (h4writer) [inline database/migration-1.php]: Don't use the id to remove it, but check for some unique data like e.g. name to delete the right row.

**Comment by h4writer** [comment]
Can you also create a subfolder "website/servo" where you can put the adjusted awfy.js and index.html for servo? Else I cannot merge this code upstream ;)

File: database/migration-1.php (hunk 2 above)
Comment (JayNakrani) [inline database/migration-1.php:17]: Done.

File: database/migration-1.php (hunk 3 above)
Comment (JayNakrani) [inline database/migration-1.php]: Done.

File: database/migration-1.php (hunk 4 above)
Comment (JayNakrani) [inline database/migration-1.php]: Done.

File: database/migration-1.php (hunk 5 above)
Comment (JayNakrani) [inline database/migration-1.php]: Done.

**Comment by JayNakrani** [comment]
Thanks! Addressed the comments. PTAL.

**Comment by JayNakrani** [comment]
See it live at [http://ec2-52-88-122-83.us-west-2.compute.amazonaws.com/servo/](http://ec2-52-88-122-83.us-west-2.compute.amazonaws.com/servo/)

**Comment by h4writer** [comment]
Thanks. LGTM

**Comment by JayNakrani** [comment]
Great. Thanks for the review.

**Pull request merged**